src/components/AttachmentPicker/index.native.js

```diff
@@ -28,54 +35,72 @@ const documentPickerOptions = {
 };
 
 /**
- * Launch the DocumentPicker. Results are in same format as ImagePicker, so we can pass the repsonse to the
- * callback as is.
+ * Inform the users when they need to grant camera access and guide them to settings
+ */
+function showPermissionsAlert() {
+    Alert.alert(
+        'ExpensifyCash does not have access to your camera. To enable access, tap Settings and turn on Camera.',
+        '',
+        [
+            {
+                text: 'Cancel',
+                style: 'cancel',
+            },
+            {
+                text: 'Settings',
+                onPress: () => Linking.openURL('app-settings:'),
+            },
+        ],
+        {cancelable: false},
+    );
+}
+
+/**
+ * A generic handling when we don't know the exact reason for an error
+ *
+ * @param {String} message
+ */
+function showGeneralAlert(message) {
+    Alert.alert(
+        'An error is preventing us to use handle the attachment',
+        message, // Todo: maybe we don't want to show this as it's probably not human friendly
+    );
+}
+
+/**
+ * Launch the DocumentPicker. Results are in the same format as ImagePicker
  *
- * @param {Object} callback
+ * @returns {Promise<DocumentPickerResponse>}
  */
-function showDocumentPicker(callback) {
-    RNDocumentPicker.pick(documentPickerOptions).then((results) => {
-        callback(results);
-    }).catch((error) => {
+function showDocumentPicker() {
+    return RNDocumentPicker.pick(documentPickerOptions).catch((error) => {
         if (!RNDocumentPicker.isCancel(error)) {
+            showGeneralAlert(error.message);
             throw error;
         }
     });
 }
 
 /**
- * Launch the AttachmentPicker. We display the ImagePicker first, as the document option is displayed as a
- * custom ImagePicker list item.
+ * Common image picker handling
  *
- * @param {Object} callback
+ * @param {function} imagePickerFunc - RNImagePicker.launchCamera or RNImagePicker.launchImageLibrary
+ * @returns {Promise<ImagePickerResponse>}
  */
-function show(callback) {
-    RNImagePicker.showImagePicker(imagePickerOptions, (response) => {
-        if (response.error) {
-            if (response.error === 'Camera permissions not granted') {
-                Alert.alert(
-                    // eslint-disable-next-line max-len
-                    'ExpensifyCash does not have access to your camera. To enable access, tap Settings and turn on Camera.',
```

Comment:
Should say: 'Expensify.cash does not have access to your camera. Please enable the permission and try again.'

Reply:
Wrong place, but you get the idea

Reply:
@Julesssss can you take a look here: as that message also won't fit: https://github.com/Expensify/Expensify.cash/pull/2114#issuecomment-810436951

The first parameter is Alert `title` and the second is alert `content`

Reply:
I've addressed this with separate title and message:

#### Android
![image](https://user-images.githubusercontent.com/12156624/113141622-9756cb80-9232-11eb-863a-6d5ff000e753.png)

#### iOS (system theme is dark)
![image](https://user-images.githubusercontent.com/12156624/113141707-b5bcc700-9232-11eb-82b6-a462824f3d6f.png)

I'll let you decide is it OK and resolve

Reply:
Will do